server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -894,11 +918,93 @@ private FlowInstanceDetailResp buildFlowInstance(List<RiskLevel> riskLevels,
         return FlowInstanceDetailResp.withIdAndType(flowInstance.getId(), flowInstanceReq.getTaskType());
     }
 
+    private FlowInstanceDetailResp buildFlowInstanceForMultipleDatabase(List<RiskLevel> riskLevels,
+            CreateFlowInstanceReq flowInstanceReq, List<ConnectionConfig> connectionConfigs) {
+        log.info("Start creating flow instance, flowInstanceReq={}", flowInstanceReq);
+        MultipleDatabaseChangeParameters parameters =
+                (MultipleDatabaseChangeParameters) flowInstanceReq.getParameters();
+        List<TaskEntity> preCheckTaskEntityList = parameters.getDatabases().stream().map(x -> {
+            CreateFlowInstanceReq preCheckReq = new CreateFlowInstanceReq();
+            preCheckReq.setTaskType(TaskType.PRE_CHECK);
+            preCheckReq.setConnectionId(x.getDataSource().getId());
+            preCheckReq.setDatabaseId(x.getId());
+            preCheckReq.setDatabaseName(flowInstanceReq.getDatabaseName());
+            return preCheckReq;
+        }).map(x -> taskService.create(x, (int) TimeUnit.SECONDS
+                .convert(flowTaskProperties.getDefaultExecutionExpirationIntervalHours(), TimeUnit.HOURS))).collect(
+                        Collectors.toList());
+        TaskEntity taskEntity = taskService.create(flowInstanceReq, (int) TimeUnit.SECONDS
+                .convert(flowTaskProperties.getDefaultExecutionExpirationIntervalHours(), TimeUnit.HOURS));
+        Verify.notNull(taskEntity.getId(), "TaskId can not be null");
+
+        FlowInstance flowInstance = flowFactory.generateFlowInstance(generateFlowInstanceName(flowInstanceReq),
+                flowInstanceReq.getParentFlowInstanceId(),
+                flowInstanceReq.getProjectId(),
+                flowInstanceReq.getDescription());
+        Verify.notNull(flowInstance.getId(), "FlowInstance id can not be null");
+
+        try {
+            FlowInstanceConfigurer startConfigurer = flowInstance.newFlowInstance();
+            for (int i = 0; i < preCheckTaskEntityList.size(); i++) {
+                FlowTaskInstance riskDetectInstance;
+                if (i == 0) {
```

Comment:
what is the difference for first pre check task ?

Reply:
The first riskDetectInstance contains the starting point